packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -61,27 +335,325 @@ library ConnextUtils {
+      // fast liquidity path
+      // credit the router the asset
+      uint256 routerAmt = amount / pathLen;
+      for (uint256 i; i < pathLen; ) {
+        _routerBalances[routers[i]][token] += routerAmt;
+        unchecked {
+          i++;
+        }
+      }
+    }
+
+    emit Reconciled(transferId, _origin, routers, token, amount, msg.sender);
+  }
+
+  /**
+   * @notice This is used by any router to decrease their available liquidity for a given asset.
+   * @param _amount - The amount of liquidity to remove for the router
+   * @param _local - The address of the asset you're removing liquidity from. If removing liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   * @param _recipient The address that will receive the liquidity being removed
+   */
+  function removeLiquidity(
+    uint256 _amount,
     address _local,
+    address _recipient,
+    mapping(address => mapping(address => uint256)) storage _routerBalances,
+    IWrapped _wrapper
+  ) external {
+    // Sanity check: to is sensible
+    if (_recipient == address(0)) revert ConnextUtils__removeLiquidity_recipientEmpty();
+
+    // Sanity check: nonzero amounts
+    if (_amount == 0) revert ConnextUtils__removeLiquidity_amountIsZero();
+
+    uint256 routerBalance = _routerBalances[msg.sender][_local];
+    // Sanity check: amount can be deducted for the router
+    if (routerBalance < _amount) revert ConnextUtils__removeLiquidity_insufficientFunds();
+
+    // Update router balances
+    unchecked {
+      _routerBalances[msg.sender][_local] = routerBalance - _amount;
+    }
+
+    // Transfer from contract to specified to
+    _transferAssetFromContract(_local, _recipient, _amount, _wrapper);
+
+    // Emit event
+    emit LiquidityRemoved(msg.sender, _recipient, _local, _amount, msg.sender);
+  }
+
+  function xcall(
+    xCallLibArgs calldata _args,
+    mapping(address => ConnextMessage.TokenId) storage _adoptedToCanonical,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools,
+    mapping(bytes32 => uint256) storage _relayerFees
+  ) external returns (bytes32, uint256) {
+    _xcallSanityChecks(_args);
+
+    // get the true transacting asset id (using wrapped native instead native)
+    (bytes32 transferId, bytes memory message, xCalledEventArgs memory eventArgs) = _processXcall(
+      _args,
+      _adoptedToCanonical,
+      _adoptedToLocalPools
+    );
+
+    // Store the relayer fee
+    _relayerFees[transferId] = _args.xCallArgs.relayerFee;
+
+    // emit event
+    emit XCalled(transferId, _args.xCallArgs, eventArgs, _args.nonce, message, msg.sender);
+
+    return (transferId, _args.nonce + 1);
+  }
+
+  /**
+   * @notice Called on the destination domain to disburse correct assets to end recipient
+   * and execute any included calldata
+   * @dev Can be called prior to or after `handle`, depending if fast liquidity is being
+   * used.
+   */
+  function execute(
+    ExecuteLibArgs calldata _args,
```

Comment:
we require that the user pass in the `LIQUIDITY_FEE_NUMERATOR` and `LIQUIDITY_FEE_DENOMINATOR`, but it looks like we ignore these args in favor of a local constant... maybe we should separate those arguments out for now? Just pass them in directly using the local constants to `_handleExecuteLiquidity => _getFastTransferAmount`

Reply:
Not sure if I understand this one.
The `execute` on ConnextLogic (former ConextUtils) is only used by ConnextHandler, not by the user

Reply:
Yeah this was a misinterpretation on my part, which is why i resolved it haha. Ty for answer tho